source/GameInterface/Services/Armies/Data/ArmyCreationData.cs

```diff
@@ -0,0 +1,32 @@
+﻿using ProtoBuf;
+using TaleWorlds.CampaignSystem;
+using TaleWorlds.CampaignSystem.Settlements;
+
+namespace GameInterface.Services.Armies.Data;
+
+/// <summary>
+/// Data required for creating an Army
+/// </summary>
+[ProtoContract(SkipConstructor = true)]
+public record ArmyCreationData
+{
+    internal ArmyCreationData(Kingdom instance, Hero armyLeader, Settlement targetSettlement, Army.ArmyTypes selectedArmyType, string newArmyId)
+    {
+        KingdomStringId = instance.StringId;
+        LeaderHeroStringId = armyLeader.StringId;
+        TargetSettlementStringId = targetSettlement.StringId;
+        SelectedArmyType = (short)selectedArmyType;
```

Comment:
should be convert with ToString()
![image](https://github.com/Bannerlord-Coop-Team/BannerlordCoop/assets/23268707/f91c4a40-7329-42e0-9e20-47abf7115c11)
I'll fix it

Reply:
int/short casting should work, reverted